Content.Server/Traits/TraitSystem.Functions.cs

```diff
@@ -297,16 +297,52 @@ public override void OnPlayerSpawn(EntityUid uid,
 public sealed partial class TraitVVEdit : TraitFunction
 {
     [DataField, AlwaysPushInheritance]
-    public Dictionary<string, string> VVEdit { get; private set; } = new();
+    public Dictionary<string, string> Changes { get; private set; } = new();
 
     public override void OnPlayerSpawn(EntityUid uid,
         IComponentFactory factory,
         IEntityManager entityManager,
         ISerializationManager serializationManager)
     {
         var vvm = IoCManager.Resolve<IViewVariablesManager>();
-        foreach (var (path, value) in VVEdit)
-            vvm.WritePath(path, value);
+        string idPath;
+        foreach (var (path, value) in Changes)
+        {
+            idPath = path.Replace("$ID", uid.ToString());
+            vvm.WritePath(idPath, value);
+        }
+    }
+}
+
+/// Only use this if you know what you're doing. This function directly writes to any arbitrary component, relative to the current value. Only works for floats.
+[UsedImplicitly]
+public sealed partial class TraitVVModify : TraitFunction
+{
+    [DataField, AlwaysPushInheritance]
+    public Dictionary<string, float> Changes { get; private set; } = new();
+
+    [DataField, AlwaysPushInheritance]
+    public bool Multiply { get; private set; } = false; // Should the value be multiplied compared to the current one? If not, add/subtract instead.
+
+    public override void OnPlayerSpawn(EntityUid uid,
+        IComponentFactory factory,
+        IEntityManager entityManager,
+        ISerializationManager serializationManager)
+    {
+        var vvm = IoCManager.Resolve<IViewVariablesManager>();
+        float newValue;
+        string idPath;
+        foreach (var (path, value) in Changes)
+        {
+            idPath = path.Replace("$ID", uid.ToString());
+            if (!float.TryParse(vvm.ReadPathSerialized(idPath), out var currentValue))
+                continue;
+            if (Multiply)
+                newValue = currentValue * value;
+            else
+                newValue = currentValue + value;
+            vvm.WritePath(idPath, newValue.ToString());
```

Comment:
```suggestion
        foreach (var (path, value) in Changes)
        {
            var idPath = (string) path.Replace("$ID", uid.ToString());
            var newValue = 0f;
            if (!float.TryParse(vvm.ReadPathSerialized(idPath), out var currentValue))
                continue;
            if (Multiply)
                newValue = currentValue * value;
            else
                newValue = currentValue + value;
            vvm.WritePath(idPath, newValue.ToString());
```
This should also be correct.

Reply:
Updated